docs-website/topics/serialization-json-io-sources.md

```diff
@@ -0,0 +1,280 @@
+
+
+### Serialize JSON to Sinks
+
+Use the `.encodeToSink()` function to serialize JSON to a `Sink`:
+
+```kotlin
+// Imports declarations from the serialization library
+import kotlinx.serialization.*
+import kotlinx.serialization.json.*
+
+// Imports declarations for kotlinx-io types and JSON I/O support
+import kotlinx.serialization.json.io.*
+import kotlinx.io.*
+import kotlinx.io.files.*
+
+@Serializable
+data class Project(val name: String, val stars: Int)
+
+@OptIn(ExperimentalSerializationApi::class)
+fun main() {
+    val project = Project("kotlinx.serialization", 9000)
+
+    // Creates a Sink for the project.json file
+    val path = Path("project.json")
+    SystemFileSystem.sink(path).buffered().use { sink: Sink ->
+
+        // Serializes the Project instance directly into a Sink
+        Json.encodeToSink(project, sink)
+    }
+}
+```
+
+### Deserialize JSON from Sources
+
+To deserialize JSON from a `Source`, use the `.decodeFromSource()` function:
+
+```kotlin
+// Imports declarations from the serialization library
+import kotlinx.serialization.*
+import kotlinx.serialization.json.*
+
+// Imports declarations for kotlinx-io types and JSON I/O support
+import kotlinx.serialization.json.io.*
+import kotlinx.io.*
+import kotlinx.io.files.*
+
+@Serializable
+data class Project(val name: String, val stars: Int)
+
+@OptIn(ExperimentalSerializationApi::class)
+fun main() {
+    // Opens a Source for the project.json file
+    val path = Path("project.json")
+    SystemFileSystem.source(path).buffered().use { source: Source ->
+
+        // Deserializes a Project instance directly from a Source
+        val project = Json.decodeFromSource<Project>(source)
+
+        println(project)
+    }
+}
+```
+
+If your input contains a large JSON array or multiple top-level JSON objects,
+you can turn a `Source` into a lazily decoded `Sequence<T>` with the
+`.decodeSourceToSequence()` function.
+
+```kotlin
+// Imports declarations from the serialization library
+import kotlinx.serialization.*
+import kotlinx.serialization.json.*
+
+// Imports declarations for kotlinx-io types and JSON I/O support
+import kotlinx.serialization.json.io.*
+import kotlinx.io.*
+import kotlinx.io.files.*
+
+@Serializable
+data class Project(val name: String, val stars: Int)
+
+@OptIn(ExperimentalSerializationApi::class)
+fun main() {
+    // Opens a Source for the projects.json file containing multiple JSON objects
+    val path = Path("projects.json")
+    SystemFileSystem.source(path).buffered().use { source: Source ->
+
+        // Lazily deserializes each Project as it is read from the Source
+        val projects: Sequence<Project> = Json.decodeSourceToSequence(source)
+
+        for (project in projects) {
+            println(project)
+        }
+    }
+}
+```
+
+> You can iterate through sequences returned by `.decodeSourceToSequence()` only once
+> because they are tied to the underlying `Source`.
```

Comment:
Please add "Closing the stream before returned sequence is evaluated completely may result in IOException." here as well